### PR TITLE
Fixes type vars semanal with `ParamSpec`

### LINKED ISSUE #11218: using a ParamSpec as a base type causes crash
```py
from typing import ParamSpec

P = ParamSpec("P")

class MyFunction(P):
    ...
```
```
test.py:5: error: INTERNAL ERROR -- Please try using mypy master on Github:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
If this issue continues with mypy master, please report a bug at https://github.com/python/mypy/issues
version: 0.920+dev.febec7123c839fce87815dbe847266596f985d72
```

### PR DESCRIPTION
**Pull request by sobolevn**
Closes #11218

### CODE REVIEW

**Review by hauntsaninja — COMMENTED** [review]
Looks good, thanks, could you add a test for valid use of ParamSpec in generic classes? It won't work, but worth checking it doesn't crash (https://www.python.org/dev/peps/pep-0612/#user-defined-generic-classes for examples)

**Comment by sobolevn** [comment]
Yes, `mypy_primer` is here falling as well.

**Comment by hauntsaninja** [comment]
The mypy_primer failure might be unrelated, let me see...

File: test-data/unit/semanal-errors.test [hunk 1]
```diff
@@ -41,6 +41,49 @@ x = None # type: X
...
+class MyFunction(P):
+    ...
+
+a: MyFunction[int]
+[out]
+main:5: error: Invalid location for ParamSpec "P"
+main:5: note: You can use ParamSpec as the first argument to Callable, e.g., 'Callable[P, int]'
+main:8: error: "MyFunction" expects no type arguments, but 1 given
+
+[case testInvalidParamSpecType3]
```
Comment (hauntsaninja) [inline test-data/unit/semanal-errors.test:73]:
```suggestion
[case testGenericParamSpec]
```

**Review by JukkaL — APPROVED** [review]
Thanks for the PR!

**Pull request merged**